### PR TITLE
fix: remove use of echo for snyk code flow

### PR DESCRIPTION
**Pull request by scott-es**
- [X] follow CONTRIBUTING rules
- [X] Detailed description:
This fix PR removes use of the system `echo` command which is causing an error running on runners that don't have this command available.  [Here](https://snyk.zendesk.com/agent/tickets/70704) is a ticket describing a customer issue. 
- [X] Risk assessment: High
- [X] Non-breaking API
- [X] Test-units
- [X] No changes required at Gitbook

**User facing Documentation**
- [X] No other Gitbook PR documentation required
- [X] commit to prefix with fix:
- [X] No CHANGELOG.md present

**Testing**
- [X] testcases must pass in configured CI Job

### CODE REVIEW

File: snykTask/src/index.ts [hunk 1]
```diff
@@ -140,6 +140,40 @@ async function showDirectoryListing(
   }
 }
 
+async function runSnykCodeTest(
```
Comment (PeterSchafer) [inline snykTask/src/index.ts:143]: Suggestion: the actual Code test is being run [here](https://github.com/snyk/snyk-azure-pipelines-task/pull/189/files#diff-7c914af7be18308e0eadff52061e050bfb1a024cfbdb6315772e2273e1b20cc8R221). This method generates a code result when no issues are found. My suggestion would be to rename it something like `generateCodeResultsWithoutIssues()` to be more precise what it is for.

File: snykTask/src/index.ts [hunk 2]
```diff
@@ -199,28 +233,15 @@ async function runSnykTest(
   }
   const snykOutput: SnykOutput = { code: code, message: errorMsg };
 
-  // handle snyk code no-issues-found non-existent json by rerunning --json stdout to piped file
+  // handle bug in snyk CLI for code test, when --json-file-output is specified
```
Comment (PeterSchafer) [inline snykTask/src/index.ts:236]: Nitpick: I know it is a nitpick, but the current behaviour is officially documented so it is more an inconsistency than a bug :)

Comment (scott-es) [reply snykTask/src/index.ts]: changed to "inconsistency"

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 version: 2.1
 
 orbs:
-  prodsec: snyk/prodsec-orb@1.0
+  prodsec: snyk/prodsec-orb@1.1
```
Comment (PeterSchafer) [inline .circleci/config.yml:4]: Question: Is there a reason for upgrading the version here?

File: snykTask/src/__tests__/task-lib.test.ts [hunk 4]
```diff
@@ -226,6 +226,41 @@ describe('getOptionsForSnykToHtml', () => {
   });
 });
 
+describe('generateSnykCodeResultsWithoutIssues sends console output to file', () => {
```
Comment (PeterSchafer) [inline snykTask/src/__tests__/task-lib.test.ts:229]: Issue: The name of the test indicates that the function `generateSnykCodeResultsWithoutIssues()` would be tested, this is not the case. instead the test replicates what the function does in a way that doesn't test the actual code. I would say that this test doesn't help to ensure the quality of the code and can be removed. 

I will commit a basic test myself, which we can use to discuss on what else needs to be tested.

File: .circleci/config.yml (hunk 3 above)
Comment (gwnlng) [reply .circleci/config.yml:4]: yes, the pipeline _Scan for hardcoded secrets_ job fails without the orb upgrade on some `git error encountered, see logs` - git connectivity based.

File: snykTask/src/__tests__/task-lib.test.ts [hunk 5]
```diff
@@ -20,6 +20,10 @@ import * as fs from 'fs';
 import * as os from 'os';
 import * as path from 'path';
 import * as fse from 'fs-extra';
+import { generateSnykCodeResultsWithoutIssues } from '../index';
+import { getSnykDownloadInfo, downloadExecutable } from '../install';
+
+jest.setTimeout(120_000);
```
Comment (thisislawatts) [inline snykTask/src/__tests__/task-lib.test.ts:26]: **question:** What did we introduce here that made a 2 min time out required?

**Review by thisislawatts — APPROVED** [review]

File: snykTask/src/__tests__/task-lib.test.ts (hunk 5 above)
Comment (PeterSchafer) [reply snykTask/src/__tests__/task-lib.test.ts:26]: The download of the CLI in beforeAll()

**Pull request merged**